src/mon/PGMap.cc

```diff
@@ -4602,16 +4607,38 @@ int reweight::by_utilization(
       osd_util.second = (double)p.second.kb_used / (double)p.second.kb;
     }
     util_by_osd.push_back(osd_util);
+    
+    // Overweighted devices and their weights are factors to
+    // calculate reweight_emergency. One 10% underfilled device
+    // with 5 2% overfilled devices, is arguably a better
+    // situation than one 10% overfilled with 5 2% underfilled devices
+    
+    if ( osd_util.second >= average_util){
+      unsigned weight = osdmap.get_weight(p.first);
+      sum_weights += weight;
+
+      // cdf of normal distribution:https://stackoverflow.com/a/18786808
+      reweight_urgency +=
+	weight * (erfc(-((osd_util.second - average_util)/average_util) * M_SQRT1_2)-1);
```

Comment:
1. I used a shifted and scaled version of cdf of standard normal distribution.

2. I wanted to have a function F(x), where x = `(osd_util.second - average_util) / average_util`, which is bounded between 0 and 1, so that ultimately `reweight_urgency` will also be bounded. A  larger value of x, should imply more urgency to reweight. Also, the difference between F(x) when x is large, should be minimal. The value of F(x) should get close to 1 (highest urgency to reweight) with steeply. I had also thought of using F(x) = (1 - e^(-x)). But that had slower convergence to 1, compared to the function currently in use.
 
3. Since I am only considering the over-utilized devices, so
`(osd_util.second - average_util) / average_util` will be greater than 0. The  \Phi(0) = 0.5 . I shifted the function by 0.5 and scaled it by 2 times, so that, F(0) = 0 and F(x) asymptotically tends to 1. So, the function I am using is 2*(Phi(x) - 0.5). I will include it in a comment.